src/prelude/internals.mo

```diff
@@ -75,8 +75,16 @@ func @mut_array_vals<A>(xs : [var A]) : () -> @Iter<A> =
     let l = xs.size();
     public func next() : ?A { if (i >= l) null else {let j = i; i += 1; ?xs[j]} };
   };
-func @blob_size(xs : Blob) : () -> Nat =
-  func () : Nat = (prim "blob_size" : Blob -> Nat) xs;
+func @blob_size(b : Blob) : () -> Nat =
+  func () : Nat = (prim "blob_size" : Blob -> Nat) b;
+func @blob_keys(b : Blob) : () -> @Iter<Nat> =
+  func () : @Iter<Nat> = object {
```

Comment:
```suggestion
  class () {
```
writing like this should work, but it fails in the IR-typechecker:
```
Ill-typed intermediate code after Desugaring (use -v to see dumped IR):
internals:81.3-85.4: IR type error [M0000], subtype violation:
  ()
  () -> @Iter<Nat>
```
but why?

Reply:
Interesting, maybe we are desugaring class decs incorrectly when used as an expression.
Maybe construct a repo and file as a separate issue.

Reply:
See #5050, repro in b078e8e0d0229addc7c423b3e55719000ac40cd1 and 72eba0363848b1bfe14ce530b776b35b5e2368ab.